### PR TITLE
feat: introduce VotesERC20Lockable contract

### PR DESCRIPTION
**Pull request by adamgall**
Closes ENG-568

# Add VotesERC20LockableV1 Token Contract with Comprehensive Testing

This PR introduces a new lockable version of our VotesERC20 token contract with extensive test coverage. The implementation allows for controlled token transfers through a locking mechanism and whitelist system.

## Core Features

- **Token Locking**: Owner can lock/unlock token transfers globally
- **Whitelist System**: Addresses can be whitelisted to transfer tokens even when locked
- **Owner Privileges**: 
  - Can transfer tokens regardless of lock status
  - Can mint new tokens at any time
  - Manages whitelist and lock status
- **ERC165 Support**: Full interface detection for ILockableV1, IMintableV1, and other interfaces

### New Contracts
- `VotesERC20LockableV1.sol`: Main implementation extending VotesERC20
- `ILockableV1.sol`: Interface defining locking and whitelist functionality
- `IMintableV1.sol`: Interface for minting capabilities

### Key Implementation Points
- Uses OpenZeppelin's ERC165Storage for interface detection
- Implements `isTransferable` modifier for transfer control
- Supports proxy initialization with flexible parameters
- Version tracking with constant VERSION = 1

### Security Features
- All privileged functions are owner-restricted
- Events emitted for lock status changes and whitelist updates
- Comprehensive checks for lock/unlock state transitions
- Safe initialization pattern with `_disableInitializers()`

## Testing

Extensive test suite covering:
- Initialization scenarios
- Locking/unlocking functionality
- Whitelist management
- Token transfers (direct and approved)
- Minting operations
- Interface support (ERC165)
- Edge cases and access control

## Security Considerations

- Only owner can lock/unlock tokens and manage whitelist
- Minting restricted to owner regardless of lock status
- All state-changing functions emit appropriate events
- Initialization can only occur once
- Transfer restrictions properly enforced through modifier

### CODE REVIEW

**Review by johnqh — APPROVED** [review]

**Comment by adamgall** [comment]
I still have some more cleanup and ERC165 support to add to this contract, let's not merge yet

edit: done

**Review by dacebt — COMMENTED** [review]
reviewed. no additional comments. looks good to me. I'll revisit when you finish cleanup

File: contracts/VotesERC20LockableV1.sol [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import {ERC165Storage} from "@openzeppelin/contracts/utils/introspection/ERC165Storage.sol";
+
+contract VotesERC20LockableV1 is ILockableV1, IMintableV1, VotesERC20, Version {
+    uint16 private constant VERSION = 1;
+
+    bool public locked;
+    mapping(address => bool) public whitelisted;
+
+    event Locked(bool locked);
+    event Whitelisted(address indexed account, bool isWhitelisted);
```
Comment (twodam) [inline contracts/VotesERC20LockableV1.sol]: move events into interface `ILockableV1`?

**Review by twodam — APPROVED** [review]

File: contracts/VotesERC20LockableV1.sol (hunk 1 above)
Comment (adamgall) [reply contracts/VotesERC20LockableV1.sol]: Good call, will do this

Comment (adamgall) [reply contracts/VotesERC20LockableV1.sol]: Moved those events, and created custom errors, into the interface
